pms-115-120/utility/src/main/java/io/mosip/pms/ida/service/PMSDataMigrationService.java

```diff
@@ -164,4 +245,29 @@ private JSONObject getPolicyObject(String policy) {
 		}
 		return null;
 	}
+	
+	public String getUser() {
+		if (Objects.nonNull(SecurityContextHolder.getContext())
+				&& Objects.nonNull(SecurityContextHolder.getContext().getAuthentication())
+				&& Objects.nonNull(SecurityContextHolder.getContext().getAuthentication().getPrincipal())
+				&& SecurityContextHolder.getContext().getAuthentication().getPrincipal() instanceof AuthUserDetails) {
+			return ((AuthUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal())
+					.getUserId();
+		} else {
+			return "system";
+		}
+	}
+
+	public LocalDateTime getLastSyncTimeStamp() {
+		LastSync lastSyncDto = lastSyncRepository.findLastSync();
+		return (lastSyncDto == null) ? null : lastSyncDto.getLastSync();
+	}
+	
+	public void saveLatestSyncTimeStamp(LocalDateTime LatestSync) {
+		LastSync latestSync = new LastSync();
+		latestSync.setLastSync(LatestSync);
+		latestSync.setCreatedBy(getUser());
+		latestSync.setCreatedDateTime(LatestSync);
+		lastSyncRepository.save(latestSync);
```

Comment:
If we keep creating new records, this table will keep growing. Can't we update a single record rather ?

Reply:
we can do that. I'll update the code and push it sir.